# The pangenome of the wheat pathogen Pyrenophora tritici-repentis reveals novel transposons associated with necrotrophic effectors ToxA and ToxB
Ryan Gourlieᵃ, Megan McDonaldᵇ, Mohamed Hafezᵃ, Rodrigo Ortega-Poloᵃ, Kristin E. Lowᵃ, D. Wade Abbottᵃ, Stephen E. Strelkovᶜ, Fouad Daayfᵈ and Reem Aboukhaddourᵃ*

ᵃAgriculture and Agri-food Canada, Lethbridge, Alberta, Canada; bUniversity of Birmingham, School of Biosciences, Edgbaston, Birmingham, United Kingdom; ᶜUniversity of Alberta, Faculty of Agricultural, Life, and Environmental Sciences, Edmonton, Alberta, Canada; ᵈUniversity of Manitoba, Faculty of Agricultural and Food Sciences, Winnipeg, Manitoba, Canada

Link: https://www.biorxiv.org/content/10.1101/2022.03.07.483352v1

Background:
In fungal plant pathogens, genome rearrangements followed by selection pressure for adaptive traits
have facilitated the co-evolutionary arms race between hosts and their pathogens. Pyrenophora tritici-
repentis (Ptr) has emerged recently as a foliar pathogen of wheat worldwide and its populations consist
of isolates that vary in their ability to produce combinations of different necrotrophic effectors. These
effectors play vital roles in disease development. Here, we sequenced the genomes of a global collection
(40 isolates) of Ptr to gain insights into its gene content and genome rearrangements.

Results:
A comparative genome analysis revealed an open pangenome, with an abundance of accessory genes (~
57%) reflecting Ptr’s adaptability. A clear distinction between pathogenic and non-pathogenic genomes
was observed in size, gene content, and phylogenetic relatedness. Chromosomal rearrangements and
structural organization, specifically around effector coding genes, were detailed using long-read
assemblies (PacBio RS II) generated in this work in addition to previously assembled genomes. We also
discovered the involvement of large mobile elements associated with Ptr’s effectors: ToxA, the gene
encoding for the necrosis effector, was found as a single copy within a 143-kb ‘Starship’ transposon
(dubbed ‘Horizon’) with a clearly defined target site and target site duplications. ‘Horizon’ was located
on different chromosomes in different isolates, indicating mobility, and the previously described
ToxhAT transposon (responsible for horizontal transfer of ToxA) was nested within this newly identified
Starship. Additionally, ToxB, the gene encoding the chlorosis effector, was clustered as three copies on a
294-kb element, which is likely a different putative ‘Starship’ (dubbed ‘Icarus’) in a ToxB-producing
isolate. ToxB and its putative transposon were missing from the ToxB non-coding reference isolate, but
the homolog toxb and ‘Icarus’ were both present in a different non-coding isolate. This suggests that
ToxB may have been mobile at some point during the evolution of the Ptr genome which is contradictory
to the current assumption of ToxB vertical inheritance. Finally, the genome architecture of Ptr was
defined as ‘one-compartment’ based on calculated gene distances and evolutionary rates.

Conclusions:
These findings together reflect on the highly plastic nature of the Ptr genome which has likely helped to
drive its worldwide adaptation and has illuminated the involvement of giant transposons in facilitating
the evolution of virulence in Ptr
